### ISSUE TITLE
timestamps

### ISSUE THREAD

**Issue by netzahdzc** [smartsdk/ngsi-timeseries-api#42]
When I include this attribute to the json structure:

```
"dateModified": {
    "value": "2017-01-18T20:45:42.697Z-0800"
  }
```

This error emerges:
```
crate.client.exceptions.ProgrammingError: SQLActionException[ColumnValidationException: Validation failed for time_index: {"metadata"={}, "type"='Text', "value"='2017-08-26T21:43:33.00Z'} cannot be cast to type timestamp]
207.249.127.152 - - [28/Dec/2017 20:30:54] "POST /notify HTTP/1.1" 500 -
INFO:werkzeug:207.249.127.152 - - [28/Dec/2017 20:30:54] "POST /notify HTTP/1.1" 500 -
```

**Comment by netzahdzc** [comment]
I just noticed that the error is exclusive of the attribute: dataModified. It vanish when I use "dateCreated" instead. Still, don't know whether this is something that could trigger future errors or create confusion in the future.

**Comment by juliozinga** [comment]
Can you explain the steps performed until you find the error?

**Comment by chicco785** [comment]
hi @netzahdzc,
@taliaga can help better, but i suppose the following, unless you don't attach a type to the data structure, it is assumed to be of type "Text", the column in the database is already a timestamp, so it returns an error.

can you try with:
```
"dateModified": {
    "value": "2017-01-18T20:45:42.697Z-0800",
    "type": "DateTime"
  }
```

Cheers,
Federico

**Comment by netzahdzc** [comment]
Hi @chicco785 , I was using the data type (i.e., DateTime); also tried with Text. I think it has to be with a private word (i.e., dateModified), because when I use dateCreated everything works well. 

Anyways, this is not something extremely urgent (since I workaround by using dateCreated instead), I just wanted to mention "the potential bug" to prevent future issues with the component.

Btw, @juliozinga, the error pop up when posting any json structure to the component QL. And I found the error by respective log.

Cheers!!

**Comment by chicco785** [comment]
let's wait for @taliaga check.

**Comment by taliaga** [comment]
Thanks @netzahdzc for raising the discussion. Let's see if we can clarify a bit and maybe we spot something to fix in QL.

First of all, I assume you are not using Orion but directly inserting to QL, right?

Since QL consumes Orion notifications and `dateModified` is the keyword used by Orion, this is the keyword QL looks for the default "time index". (note there could be many DateTime attributes, so QL needs a default for its time index). This `dateModified` is expected to be of the type `DateTime`. The error there was raised because an insert of `dateModified` tried to give type Text to it.

That's my first impression, but if you have a couple of scripts to reproduce this, I can give it a deeper look. Note that from Orion this metadata (`dateModified`) can come at the level of the entity and/or for each of the attributes. Where are you using it?

Regarding your workaround, note `dateCreated` is also a keyword in Orion (you're not having an error because QL does not parse it for now), but I suggest we try to spot the real issue before feeling safe with the workaround.

**Comment by netzahdzc** [comment]
That's right. I'm posing directly to QL.

ok about the dateModified.

I was using dateModified as attribute not metadata.

To solve the issue, Im using dateCreated instead of dataModified, but, ok about the heads-up. I will try to change it to use a different attribute date instead of those.

Btw, feel free to cose this issue at any time. I just opened to notified about what I experienced. 

Cheers!!
